File: problems/125_valid_palindrome/Solution.java
/**
 * Two sum solution
 * @author Román
 * @see {@link https://github.com/RomanKornyeyev} - GitHub profile
 */

class Solution {
    public boolean isPalindrome(String s) {
        //for default, the answer is true
        boolean answer = true;
		
        //lower case, remove the lateral spaces, remove all non-alphanumeric characters
		s = s.toLowerCase().trim().replaceAll("[^a-zA-Z0-9]", "");
		
        //if the length is 0, it returns true, otherwise it check the string
        //compare for extreme characters, going towards the center
        if(s.length() > 0)
		    for(int i=0, j=s.length(); i<((s.length()/2)+1); i++, j--)
	 		    if(!s.substring(i, i+1).equals(s.substring(j-1, j)))
	 			    answer = false;
	 			
		return answer;
    }
}

File: problems/125_valid_palindrome/Solution.py
# Author: Román
# GitHub profile: https://github.com/RomanKornyeyev

import re

class Solution(object):
    def isPalindrome(self, s):
        res = True

        # Remove non-alphanumeric characters
        # Convert the string to lower case and remove the spaces from the start and end
        s = re.sub("[^a-zA-Z0-9]", "", s)
        s = s.lower().strip()

        # Check if it is a palindrome by comparing characters from the start and end
        if len(s) > 0:
            for i in range(0, (len(s) // 2) + 1):
                if s[i] != s[len(s) - 1 - i]:
                    res = False # If they don't match, change the result to false
        
        return res